testng-core/src/test/java/org/testng/internal/invokers/ParameterHandlerTest.java

```diff
@@ -66,8 +64,7 @@ private void runTest(ITestNGMethod testNGMethod, ParameterHolder.ParameterOrigin
     assertThat(params.parameterHolder).isNotNull();
     assertThat(params.parameterHolder.origin).isEqualByComparingTo(origin);
     Iterator<Object[]> iterators = params.parameterHolder.parameters;
-    assertThat(Lists.newArrayList(iterators))
-        .containsAll(Collections.singletonList(new Object[] {"bar"}));
+    assertThat(iterators).toString().equals("bar");
```

Comment:
I have some doubts about this line. It will return a boolean, right?

Reply:
It asserts that iterators.toString() == "bar" - feel free to change it.
The original contained the .containsAll() which threw an error in JDK17.

Reply:
@hduerkop - This assertion is incorrect.  Can you please fix it ?

The below sample should clarify.

```java
  @Test
  public void testMethod() {
    Iterator<String> actual = Collections.singletonList("bar").iterator();
    assertThat(actual.toString()).isEqualTo("bar"); // This does an assertion and we should have done this

    // This is just returning true or false and is not being consumed by any assertion and we are doing this in this PR
    // Here we are not invoking any of the method provided by assertj but instead invoking `Object.equals()`
    assertThat(actual).toString().equals("bar"); 
  }

```

Reply:
how about assertThat(params.parameterHolder.parameters.toString()).hasToString("bar"); then?

Reply:
@hduerkop - It does not work

There are test failures 

```
FAILURE   0.0sec, org.testng.internal.invokers.ParameterHandlerTest > testCreateParameters
    org.opentest4j.AssertionFailedError: 
    Expecting actual's toString() to return:
      "bar"
    but was:
      "org.testng.internal.collections.ArrayIterator@1"
        at app//org.testng.internal.invokers.ParameterHandlerTest.runTest(ParameterHandlerTest.java:65)
        at app//org.testng.internal.invokers.ParameterHandlerTest.testCreateParameters(ParameterHandlerTest.java:38)
```

Why did we have to make this change? Just curious. Our CI builds also uses JDK17 and we have not had failures around this test. Can we revert this ? 

In a nutshell, I was hoping that we confine the changeset to just dependency upgrades alone.

Reply:
version.org.assertj..assertj-core=3.11.0 works, 3.12.0 and throws:
Cannot resolve method 'containsAll' in 'IteratorAssert'

Reply:
https://assertj.github.io/doc/#assertj-core-3.12.0-breaking-changes